odl/solvers/functional/default_functionals.py

```diff
@@ -453,25 +466,36 @@ class IndicatorLpUnitBall(Functional):
     norm.
     """
 
-    def __init__(self, space, exponent):
+    def __init__(self, domain, exponent, range=None):
         """Initialize a new instance.
 
         Parameters
         ----------
-        space : `DiscreteLp` or `TensorSpace`
+        domain : `DiscreteLp` or `TensorSpace`
             Domain of the functional.
         exponent : int or infinity
             Specifies wich norm to use.
+        range : `Field`, optional
+            Range of the functional. The default `None` will be treated as
+            `domain.field`.
         """
-        super(IndicatorLpUnitBall, self).__init__(space=space, linear=False)
-        self.__norm = LpNorm(space, exponent)
+        super(IndicatorLpUnitBall, self).__init__(domain=domain, linear=False,
+                                                  range=range)
+
+        self.__norm = LpNorm(domain, exponent, range=RealNumbers())
         self.__exponent = float(exponent)
+        self.__range = range
 
     @property
     def exponent(self):
         """Exponent corresponding to the norm."""
         return self.__exponent
 
+    @property
+    def range_(self):
```

Comment:
This is not needed. `IndicatorLpUnitBall` inherits from `Operator` and hence already has a `range` property, which is the one that should be used.